src/main/java/ch/njol/skript/expressions/ExprBlocks.java

```diff
@@ -122,7 +133,20 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	public Iterator<Block> iterator(final Event e) {
 		try {
 			final Expression<Direction> direction = this.direction;
-			if (direction != null) {
+			if (pattern == 4) {
+				List<Block> blocks = new ArrayList<>();
+				for (Chunk chunk : (Chunk[]) from.getAll(e)) {
+					int maxHeight = chunk.getWorld().getMaxHeight();
+					for (int x = 0; x < 16; x++) {
+						for (int z = 0; z < 16; z++) {
+							for (int y = 0; y < maxHeight; y++) {
+								blocks.add(chunk.getBlock(x, y, z));

```

Comment:
the point of the iterator method is that things are not entirely computed up front. this should be an iterator that *only* computes the next needed block

Reply:
Ill be honest, I have no clue how to do that. 
I'm going to close this PR for now until I can think of a better way to handle this.